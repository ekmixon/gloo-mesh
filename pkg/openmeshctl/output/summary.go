package output

import (
	"sort"
	"strings"

	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
)

// Summary holds human readable information about a resource.
type Summary struct {
	// Meta for the object the summary is for.
	Meta metav1.Object

	// Fields additional fields to print for the object beyond the meta
	Fields FieldSet
}

// FieldSet is an ordered set of fields
type FieldSet []Field

// AddField adds a field to the set.
func (fs *FieldSet) AddField(label string, value interface{}) {
	*fs = append(*fs, Field{Label: label, Value: value})
}

// Sort sorts the fields in the event that they are unordered to enforce order.
// For example, if a field set is built from a map, Sort can be used to add a guaranteed order.
func (fs *FieldSet) Sort() {
	sort.Slice(*fs, func(i, j int) bool {
		return strings.Compare((*fs)[i].Label, (*fs)[j].Label) < 0
	})
}

// Field is a field that can be added to the summary
type Field struct {
	Label string
	Value interface{}
}
