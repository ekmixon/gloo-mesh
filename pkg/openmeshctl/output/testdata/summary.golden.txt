Name:               details-bookinfo-mgmt-cluster
Namespace:          gloo-mesh
CreationTimestamp:  Tue, 31 Aug 2021 21:00:36 +0000
Labels:             cluster.discovery.mesh.gloo.solo.io=mgmt.cluster
                    cluster.multicluster.solo.io=
                    owner.discovery.mesh.gloo.solo.io=gloo-mesh-mgmt-cluster
Annotations:        <none>
Kube Service:
 External Addresses:  <none>
 IP Addresses:
   10.244.0.7
Traffic Policies:  <none>
Access Policies:
 gloo-mesh.details
 gloo-mesh.details-admin
Mesh:        gloo-mesh.istiod-istio-system-mgmt-cluster
Local FQDN:  details.bookinfo.svc.cluster.local
